Revised Support Hero handbook page (#7433)
PostHog/posthog.com: contents/handbook/engineering/support-hero.md

showTitle: true
---

Every week, one person in each engineering team is designated the Support Hero. If this is you this week, congratulations!

As Support Hero, your job is to investigate and resolve issues reported by customers. A single case of suspicious data or a show-stopping bug can really undermine one's confidence in a data product, so it's important that we get to the bottom of all issues.

You'll see some teams using a term of endearment for Support Hero, examples being "Infra Hero" or… "Luigi". Don't ask – we don't know.

[Marcus](https://posthog.com/community/profiles/1036), our Support Engineer, triages tickets for the Product Analytics and Pipeline team, due to the high volume of tickets those teams get. He will resolve tickets if possible, and escalate to the engineering team responsible if he needs further help.

## When is my turn?

Most engineering teams run a PagerDuty schedule:

- [Product Analytics](https://posthog.pagerduty.com/schedules#PXUZ9XL)
- [Feature Success](https://posthog.pagerduty.com/schedules#P04FUTJ)

> - [Replay](https://posthog.pagerduty.com/schedules#PUPERAV)

Reviewer: This link is broken, maybe https://posthog.pagerduty.com/schedules#PLGXQIF is the new one?

Author: Oh yeah
```suggestion
- [Replay](https://posthog.pagerduty.com/schedules#PLGXQIF)
```

- [Pipeline](https://posthog.pagerduty.com/schedules#PM8YSH8)
- [Infrastructure](https://posthog.pagerduty.com/schedules#P78OOWZ)
- [Growth](https://posthog.pagerduty.com/schedules#PN1Q6BO)

> The schedules consist of contiguous blocks, but that definitely doesn't mean working 24/7 – you should just work your normal hours.

Reviewer: Can we write down how Saturdays and Sundays should be since its not 
"normal hours" but part of the schedule anyway.

Author: Makes sense to clarify!
```suggestion
The schedules consist of contiguous blocks, but you should just work your normal hours. Support Hero is Monday to Friday only.

Reviewer: Thanks, I am part of the `All team escalation` schedule, also for the weekend, I believe All team escalation != Support Hero schedule then?

Author: Yes, that's on-call – they're very much separate processes. I renamed "All team escalation" so that this is clearer:
<img width="759" alt="Screenshot 2024-01-04 at 15 22 04" src="https://github.com/PostHog/posthog.com/assets/4550621/9f21d4c1-4440-4534-b943-87490af6a17e">
As for on-call, there's a separate PR cleaning it up, hopefully making it clearer: #7432

## What do I do as Support Hero?

Each engineering team has its own list of tickets in Zendesk:

- [Product Analytics](https://posthoghelp.zendesk.com/agent/filters/17989255082139) (escalated tickets only)
- [Feature Success](https://posthoghelp.zendesk.com/agent/filters/14507099125531)
- [Replay](https://posthoghelp.zendesk.com/agent/filters/14507048415771)
- [Pipeline](https://posthoghelp.zendesk.com/agent/filters/14506794017051)
- [Infrastructure](https://posthoghelp.zendesk.com/agent/filters/14507148758939)
- [Auth & Billing, handled by Growth](https://posthoghelp.zendesk.com/agent/filters/14507107058843)

Your job is simple: resolve ticket after ticket from your team's list.

There are three sources of tickets:

1. In-app bug reports/feedback/support tickets sent from the [Support panel](https://us.posthog.com/home#panel=support). They include a bunch of useful links, e.g. to the admin panel and to the relevant session recording.
1. Community questions asked on PostHog.com.
1. Slack threads that have been marked with the 🎫 reaction in customer support channels.

> Some of them will ask for new features. If the feature would be widely useful for users matching [our ICP](/handbook/who-we-are-building-for), it might be a good idea to build it. Otherwise, feel free to just create a feature request issue in GitHub or +1 on an existing one – you can then send a link to the user, giving them a way of tracking progress.

Reviewer: You can't +1 twice unless you add comments which is not ideal.
what's the current status quo for feature requests? how would you know if 1 or 10 customers asked for it via tickets instead of +1 themselves?

Author: I think comments are the way, here's one way to put this:
```suggestion
Some of them will ask for new features. If the feature would be widely useful for users matching [our ICP](/handbook/who-we-are-building-for), it might be a good idea to build it. Otherwise, feel free to just create a feature request issue in GitHub or, if one already exists, comment with "+1 from <ticket link>" – you can then send an issue link to the user, giving them a way of tracking progress.
```

Reviewer: I'm not a big fan of +1 comments, you still have to count them every time you wanna know how many people asked for it and every person subscribed to the issugetset alerted, but if we rely on GH only, I don't know a better way.

Reviewer 2: We are also tracking feature requests in Vitally.

```suggestion
Some of them will ask for new features. If the feature would be widely useful for users matching [our ICP](/handbook/who-we-are-building-for), it might be a good idea to build it. Otherwise, feel free to just create a feature request issue in GitHub or +1 on an existing one – you can then send a link to the user, giving them a way of tracking progress. Also make sure to let the [Customer Success Team](https://posthog.com/handbook/small-teams/customer-success) know, since they will track feature requests for paying customers.
```

cc: @simfish85

> 1. [Community questions](https://posthog.com/questions/)

Reviewer 2: ```suggestion
```

Community questions will be correctly tagged in Zendesk already. Additionally, it doesn't make sense to look at posthog.com/questions since you'll encounter a bunch of tickets that are not related to your team anyway.

Author: Good point, removed

> 1. Open Zendesk tickets for your team that have `low` priority (non-paying users)

Reviewer 3: One thing to highlight from the Support RFC is that all tickets will come in as Normal or High priority with the aim that whoever is in this role will at least look at every ticket to see if there are any patterns.  If they haven't been able to get to these then those which are non-paying will be moved to low priority after 5 days.  So this prioritization is right, we just need folks to know that there will be no newly created Low tickets.
